### ISSUE TITLE
mode.predict freezes

### ISSUE THREAD

**Issue by shubham9630** [keras-team/keras#15941]
**System information**.
- I have used tf.data.Dataset.from_generator for generating data
- OS Platform and Distribution (Windows 10):
- TensorFlow version: 2.4.1
- Python version:  3.9.5

model.predict freezes when use tf.data.dataset
The below code can be used to reproduce the problem.

def tf_data_generator(file_list, batch_size = 20):
    i = 0
    while True:
        if i*batch_size >= len(file_list):  
            i = 0
            np.random.shuffle(file_list)
        else:

            file_chunk = file_list[i*batch_size:(i+1)*batch_size]
    
            data = []
            labels = []
            
            for file in file_chunk:
                file=file.decode()
                file_name=file.split("New//")[1].split(".mat")[0]
                #print(file_name)
                idx=list_of_sz.loc[list_of_sz.iloc[:,3].str.contains(file_name, na=False)].index[0]
                temp = np.load(file)
                temp=temp['arr_0']
                temp[temp == -inf]=0
    
                if np.any(np.isnan(temp)):
                    continue
    
    
                data.append(temp)
    
                labels.append(list_of_sz.iloc[idx,1])
            data = np.asarray(data).reshape(-1,50,95000,1)
            labels = np.asarray(labels)
            yield data, labels
            i = i + 1

test_file_names=[r"/scratch/agarwase/Data/Test_Data_New//" + i for i in test_file_names]
batch_size = 8
test_dataset = tf.data.Dataset.from_generator(tf_data_generator, args = [test_file_names, batch_size],
                                             output_shapes = ((None,50,95000,1),(None,)),
                                             output_types = (tf.float32, tf.float32))

steps = len(test_file_names)//batch_size
model=models.load_model('model_trial14.h5')
model.summary()
test_pred= model.predict(test_dataset)

I tried removing the tf.Data.dataset it works with numpy array. It was working before but now it just freezes forever.

**Comment by jvishnuvardhan** [comment]
@shubham9630 Can you please share a simple standalone code to reproduce the issue? Thanks!
